audacity/audacity
# 5891 non destructive pitch shifting

Resolves: #5891, #5863

Non-destructive pitch shifting


Recommended:

## QA
Please be aware of #5728, which this PR does _not_ fix.
- [x] resolves #5863
- [x] "Change Speed" dialog is now called "Pitch And Speed"
- [x] "Pitch And Speed" dialog is accessible like before (Select clip + right-click, right-click on clip header, `Edit > Audio Clips > Pitch and Speed...`)
- [x] "Pitch And Speed" is now also accessible by "Alt+P" when clip is fully selected
- [x] "Alt+Up" / "Alt+Down" change clip pitch by +/- 1 semitone
- [x] Pitch shift can be changed also during playback
- [x] Using extreme pitch shift and speed combinations does not freeze or crash Audacity (there may be pauses in audio, but normal audio should be recoverable by coming back to more reasonable settings)
- [x] Pitch-shift by cents is available in dialog
- [x] Pitch-shift and speed changes are drawn sensible in the clip header, or not drawn at all when space doesn't suffice.
- [x] Also with language set to e.g. Arabic or Hebrew
- [x] Quitting the dialog by ESC or Cancel restores previous pitch
- [x] Changing pitch adds an entry in history and is undoable
- [x] Changing pitch makes project dirty
- [x] Pitch changes persist after closing and re-opening project
- [x] Changing pitch, saving project and re-opening with earlier Audacity (e.g. 3.4.2) works fine (no pitch shifting)
- [x] Opening older project works fine

For mix-and-rendering, copy/paste or merging clips, the same rules apply for pitch-shifted and time-stretched clips. Here are some of the items to verify I can remember:

- [x] Mix-And-Render renders pitch shift
- [x] Export audio renders pitch shift
- [x] Merging clips with different pitch shift renders them
- [x] Pasting clip of pitch shift _x_ onto other clip with pitch shift _y_ doesn't merge them
